### PR TITLE
Add support for cracking SAP's PSE files

### PR DESCRIPTION
**Pull request by kholia**
This is related to https://github.com/magnumripper/JohnTheRipper/issues/3302 (Add support for cracking SAP's PSE files with PBES1-3DES-SHA).

### CODE REVIEW

**Comment by kholia** [comment]
@claudioandre-br Hey, would it be possible for you to review (and possibly approve) this PR?

File: src/sap_pse_fmt_plug.c [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+#include "common.h"
+#include "formats.h"
+#include "loader.h"
+#include "simd-intrinsics.h"
+#include "pkcs12.h"
+#include "memdbg.h"
+
+#define FORMAT_LABEL            "sappse"
+#define FORMAT_NAME             ""
+#define ALGORITHM_NAME          "PKCS12 PBE SHA1 " SHA1_ALGORITHM_NAME
```
Comment (claudioandre-br) [inline src/sap_pse_fmt_plug.c]: Isn't missing information about DES?

File: run/pse2john.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    parser.add_option("-o", "--output", help="Filename to write the output to [stdout]")
+
+    options, args = parser.parse_args()
+
+    return options, args
+
+
+def parse_pse(filename):
+    """Parses a PSE file and produces """
+    with open(filename, "rb") as f:
```
[the remark this answers is not in the document]
Comment (kholia) [reply run/pse2john.py]: The original code (not mine) relies on the default exception handler to inform the user about errors. This seems OK to me.

File: src/sap_pse_fmt_plug.c (hunk 1 above)
Comment (kholia) [reply src/sap_pse_fmt_plug.c]: Ack. Fixed now.

**Comment by kholia** [comment]
I have fixed the problems found in the review.

**Pull request merged**